src/libraries/Common/src/Interop/Windows/Ole32/Interop.IStream.cs

```diff
@@ -62,5 +63,55 @@ internal interface IStream
 
             IStream Clone();
         }
+
+        // This needs to be duplicated from IStream above because ComWrappers doesn't support re-using
+        // ComImport interfaces. The above interface can be deleted once all usages of the built-in COM support
```

Comment:
I do not understand this problem. Could you please shed some light on this for my education? What was the failure you saw when it used IStream?

Reply:
This was something that @eerhardt brought up in some early testing. Two things were encountered:

1) Putting a `static readonly Guid` field on an interface that is marked with `ComImport`, the following error occurs:

    > `System.TypeLoadException: 'Could not load type 'IStream' from assembly 'ComWrappersIDispatch, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' because the method '.cctor' has no implementation (no RVA).'`

1) When calling `Marshal.QueryInterface()` for the interface marked with `ComImport` a failure to QI would occur.

Issues (1) makes sense to me because I believe the type system validates interfaces marked with `ComImport` are validated to not have any implementation. Issue (2) seems like there is a confusion between the two systems in some scenario but since (1) was present and the attribute really isn't needed I indicated they really are for two different scenarios so an issue could occur.

I am going to be investigating (2) in the next few days because I do want to make sure it isn't a latent bug in our `ComWrappers` implementation.

Reply:
The first problem can be solved easily by putting the Guid property somewhere else, e.g. in the ComWrappers type itself.

The second problem looks odd. It worked fine in @kant2002 's ComWrappers prototype injected via `ComWrappers.RegisterForMarshalling` (https://github.com/kant2002/WinFormsComInterop/blob/main/WinFormsComInterop/WinFormsComWrappers.cs#L177).

Reply:
> It worked fine in @kant2002 's ComWrappers prototype

I see. I also ran through some scenarios locally with `ComImport` set on some interfaces. I couldn't trigger a failure either.

Reply:
I was running into issues at first, so I split the interface which seemed to fix it (as far as I could tell). I changed it back to reusing the existing ComImport `IStream` interface and everything is working now. So I must have had some other issue.

I've removed the duplicated interface.